### PR TITLE
finished backend exercise

### PR DESCRIPTION
**Pull request by abjjabjj**
interdasting

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -40,6 +40,59 @@ def mirror(name):
...
+def delete_user(id):
+	return create_response(db.deleteById('users', id), message = 'Deleted user ' + str(id))
+
+# entrypoint for /users/<id> HTTP request
+# regardless of request type, returns error message if id not found
+@app.route('/users/<id>', methods = ['GET', 'PUT', 'DELETE'])
+def user(id):
+	matched = db.getById('users', int(id))
+	if matched is None:
+		return create_response(matched, 404, 'You\'re a noob, gimme a real user')
```
Comment (vrunjeti) [inline app.py]: digging the messages

File: app.py [hunk 2]
```diff
@@ -40,6 +40,59 @@ def mirror(name):
...
+	matched = all_users if team is None else [i for i in all_users if i['team'] == team]
+	return create_response(matched, message = 'I got you the users man')
+
+# POST /user method, requires name, age, and team to work
+# gives an error message back specifying which parameters are missing
+def post_user():
+	args = request.get_json()
+	fields = ['name', 'age', 'team']
+	user_info = [args.get(field) for field in fields]
+	if None in user_info:
```
Comment (vrunjeti) [inline app.py]: i like the way you're checking these. it's pretty concise and clear

**Review by vrunjeti — CHANGES_REQUESTED** [review]
Some overall things:

You were the first person to separate the endpoint logic functions from their declarations. Which actually got me to start writing the [Flask API Conventions Spec](https://github.com/hack4impact-uiuc/wiki/wiki/Flask-API-Conventions). Read through these, understand the _why_ sections, and update your code to reflect the conventions.

Also use (4) spaces instead of tabs pls

**Comment by abjjabjj** [comment]
Update 1 To Standards

**Pull request closed without merging**